ui/components/app/transaction-list-item/smart-transaction-list-item.component.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import ListItem from '../../ui/list-item';
+import TransactionStatus from '../transaction-status/transaction-status.component';
+import TransactionIcon from '../transaction-icon';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { formatDateWithYearContext } from '../../../helpers/utils/util';
+import {
+  TRANSACTION_GROUP_CATEGORIES,
+  TRANSACTION_GROUP_STATUSES,
+} from '../../../../shared/constants/transaction';
+
+import CancelButton from '../cancel-button';
+import { cancelSwapsSmartTransaction } from '../../../ducks/swaps/swaps';
+
+export default function SmartTransactionListItem({
+  smartTransaction,
+  isEarliestNonce = false,
+}) {
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+  const [cancelSwapLinkClicked, setCancelSwapLinkClicked] = useState(false);
+  const {
+    sourceTokenSymbol,
+    destinationTokenSymbol,
+    time,
+    status,
+  } = smartTransaction;
+  const category = TRANSACTION_GROUP_CATEGORIES.SWAP;
+  const title = t('swapTokenToToken', [
+    sourceTokenSymbol,
+    destinationTokenSymbol,
+  ]);
+  const subtitle = 'metamask';
+  const date = formatDateWithYearContext(time);
+  let displayedStatusKey;
+  if (status === 'pending') {
+    displayedStatusKey = TRANSACTION_GROUP_STATUSES.PENDING;
+  } else if (status?.startsWith('cancelled')) {
```

Comment:
I think we can use `TRANSACTION_GROUP_STATUSES .CANCELLED` here.

Reply:
Same goes for cancelled. This bit of code is supposed to pick up a cancelled status from an array of potential cancelled statuses, differentiated by them all beginning with `cancelled`. For example, some of the statuses are `cancelled_would_revert`, `cancelled_too_cheap`, `cancelled_deadline_missed`. It works here because `TRANSACTION_GROUP_STATUSES.CANCELLED` starts with the exact value but I worry that the value might change over time for either this constant or smart transactions statuses and this bit of code would break.